dbms/src/Flash/Disaggregated/DisaggregatedTask.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/Exception.h>
+#include <Common/TiFlashException.h>
+#include <Flash/Coprocessor/DAGUtils.h>
+#include <Flash/Disaggregated/DisaggregatedTask.h>
+#include <Flash/Executor/QueryExecutorHolder.h>
+#include <Flash/executeQuery.h>
+#include <Interpreters/Context.h>
+#include <Storages/DeltaMerge/Remote/DisaggSnapshot.h>
+#include <Storages/DeltaMerge/Remote/DisaggSnapshotManager.h>
+#include <Storages/DeltaMerge/Remote/DisaggTaskId.h>
+#include <Storages/DeltaMerge/Remote/Serializer.h>
+#include <Storages/Transaction/KVStore.h>
+#include <Storages/Transaction/TMTContext.h>
+#include <kvproto/disaggregated.pb.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+extern const int REGION_EPOCH_NOT_MATCH;
+} // namespace ErrorCodes
+
+DisaggregatedTask::DisaggregatedTask(ContextPtr context_, const DM::DisaggTaskId & task_id)
+    : context(std::move(context_))
+    , log(Logger::get(fmt::format("{}", task_id)))
+{}
+
+// Some preparation
+// - Parse the encoded plan
+// - Build `dag_context`
+// - Set the read_tso, schema_version, timezone
+void DisaggregatedTask::prepare(const disaggregated::EstablishDisaggTaskRequest * request)
+{
+    const auto & meta = request->meta();
+    DM::DisaggTaskId task_id(meta);
+    auto task = std::make_shared<DisaggregatedTask>(context, task_id);
```

Comment:
Why setup another `DissaggregatedTask`, maybe use stack variable is ok(like dag_req and dag_context)?